src/ngx_http_lua_ssl_module.c

```diff
@@ -303,9 +322,15 @@ ngx_http_lua_ffi_update_last_ticket_decryption_key(SSL_CTX *ctx,
     pkey = &pkey[keys->nelts-1];
 
     dd("replace the last key");
-    ngx_memcpy(pkey->name, key, 16);
-    ngx_memcpy(pkey->aes_key, key + 16, 16);
-    ngx_memcpy(pkey->hmac_key, key + 32, 16);
+    if (ngx_strlen(key) == 48){
```

Comment:
About ngx_strlen(), I got your point. But I could not a find a function that I pass the string size explicitly via arguments, however doing this: "if (key[48] == '\0')" or "if (key[80] == '\0')", works fine and is O(1). Any considerations?